modules/keyless-authentication-robot-accounts.adoc

```diff
@@ -0,0 +1,268 @@
+----
+
+. Store the `access_token` from the previously step, as it will be exchanged for a {productname} robot account token in the following procedure.
+
+[id="setting-robot-federation"]
+== Setting up a robot account federation by using the {productname} v2 UI
+
+The following procedure shows you how to set up a robot account federation by using the {productname} v2 UI.
+
+.Prerequisites
+
+* You have created an organization. The following example uses `fed_test`.
+* You have created a robot account. The following example uses `fest_test+robot1`.
+* You have configured a OIDC for your {productname} deployment. The following example uses Red Hat SSO Keycloak.
+
+.Procedure
+
+. On the Keycloak main page:
+
+.. Select the appropriate realm that is authenticated for use with {productname}. Store the issuer URL, for example, `\https://keycloak-auth-realm.quayadmin.org/realms/quayrealm`.
+
+.. Click *Users* -> the name of the user to be linked with the robot account for authentication. You must use the same user account that you used when generating the OAuth2 access token.
+
+.. On the *Details* page, store the *ID* of the user, for example, `449e14f8-9eb5-4d59-a63e-b7a77c75f770`.
+
+. On your {productname} registry:
+
+.. Navigate to *Organizations* and click the name of your organization, for example, *fed_test*.
+
+.. Click *Robot Accounts*.
+
+.. Click the menu kebab -> *Set robot federation*.
+
+.. Click the *+* symbol. 
+
+.. In the popup window, include the following information:
++
+* *Issuer URL*: `\https://keycloak-auth-realm.quayadmin.org/realms/quayrealm`, or the URL of your Keycloak realm.
+* *Subject*: `449e14f8-9eb5-4d59-a63e-b7a77c75f770`, or the *ID* of your keycloak user. 
+
+.. Click *Save*. 
+
+[id="exchanging-oauth2-robot-account-token"]
+== Exchanging an OAuth2 access token for a {productname} robot account token
+
+The following procedure leverages the `access token` generated in the previous procedure to create a new {productname} robot account token. The new {productname} robot account token is used for authentication between your OIDC provider and {productname}. 
+
+[NOTE]
+====
+The following example uses a Python script to exchange the OAuth2 access token for a {productname} robot account token.
+====
+
+.Prerequisites
+
+* You have the `python3` CLI tool installed.
+
+.Procedure
+
+. Save the following Python script in a `.py` file, for example, `robot_fed_token_auth.py`
++
+[source,python]
+----
+import requests
+import os
+
+TOKEN=os.environ.get('TOKEN')
+robot_user = "fed-test+robot1"
+
+def get_quay_robot_token(fed_token):
+    URL = "https://<quay-server.example.com>/oauth2/federation/robot/token"
+    response = requests.get(URL, auth=(robot_user,fed_token))
```

Comment:
Here in this line code, when Quay is using the self-signed SSL Cert, need add one more parameter **verify=False**

`response = requests.get(URL,auth=(robot_user,federation_token),verify=False)`

Reply:
Updated